src/Microsoft.Azure.Relay/WebSockets/NetCore/WebSocketHandle.Managed.cs

```diff
@@ -197,6 +220,226 @@ public void Abort()
             throw new WebSocketException(SR.net_webstatus_ConnectFailure);
         }
 
+        private async Task ConnectWithProxyAsync(Uri uri, ClientWebSocketOptions options, Stream stream, CancellationToken cancellationToken)
+        {
+            const string successStatusCode = "HTTP/1.1 200";
+            const string proxyAuthenticationStatusCode = "HTTP/1.1 407";
+            const string Basic = "Basic ";
+            const string Digest = "Digest ";
+            AuthenticationHeaderValue ahv = null;
+
+            string statusLine = await SendRequestToProxyAsync(uri, options, stream, ahv, cancellationToken);
+            if (statusLine == successStatusCode)
+            {
+                // Proxy connection succeeded.
+                return;
+            }
+
+            // Proxy requires authentication.
+            if (statusLine.StartsWith(proxyAuthenticationStatusCode))
+            {
+                if (statusLine.Length > proxyAuthenticationStatusCode.Length &&
```

Comment:
Would it be easier to just change `proxyAuthenticationStatusCode` to have a space at the end?